### PR TITLE
Build AWS AMI by running packer inside a container

### PR DESCRIPTION
**Pull request by bpradipt**
Tested successfully with Ubuntu.
For CentOS, SSH is failing.

### CODE REVIEW

**Comment by bpradipt** [comment]
CentOS requires the ssh_username to be set to `cloud-user`

File: aws/README.md [hunk 1]
```diff
@@ -18,15 +19,60 @@ export VPC_ID=vpc-01234567890abcdef # optional, otherwise, it creates and uses t
...
+--build-arg AWS_REGION=${AWS_REGION} \
+--build-arg BINARIES_IMG=quay.io/confidential-containers/podvm-binaries-centos-amd64 \
+--build-arg PODVM_DISTRO=centos \
+-f Dockerfile .
+```
+
 - Note down your newly created AMI_ID
 
+You can use the following CLI command as well to get the AMI_ID. The command assumes that you are using
+the default AMI name: `peer-pod-ami`
```
Comment (snir911) [inline aws/README.md:70]: maybe worth adding "once creation has completed" or something

File: aws/image/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip" && \
+     unzip awscliv2.zip && \
+     ./aws/install
+
+RUN --mount=type=secret,id=AWS_ACCESS_KEY_ID \
+    --mount=type=secret,id=AWS_SECRET_ACCESS_KEY \
+    export AWS_ACCESS_KEY_ID=$(cat /run/secrets/AWS_ACCESS_KEY_ID) && \
+    export AWS_SECRET_ACCESS_KEY=$(cat /run/secrets/AWS_SECRET_ACCESS_KEY) && \
+    cd cloud-api-adaptor/aws/image && \
+     packer init ${PODVM_DISTRO}/ && \
```
Comment (snir911) [inline aws/image/Dockerfile:64]: why packer init is needed?
(nit, also, it seems alignment here is messed)

**Review by snir911 — APPROVED** [review]
minor comments, overall LGTM ( I didn't try)

File: aws/image/Dockerfile (hunk 2 above)
Comment (bpradipt) [reply aws/image/Dockerfile:64]: without packer init, it fails to use the aws ebs builder

**Review by littlejawa — APPROVED** [review]
/lgtm

**Pull request merged**